src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,204 @@
+            'database': self.msConfig['mongoDB'],
+            'server': self.msConfig['mongoDBServer'],
+            'replicaSet': self.msConfig['mongoDBReplicaSet'],
+            'port': self.msConfig['mongoDBPort'],
+            'username': self.msConfig['mongoDBUser'],
+            'password': self.msConfig['mongoDBPassword'],
+            'mockMongoDB': self.msConfig['mockMongoDB'],
+            'connect': True,
+            'directConnection': False,
+            'logger': self.logger,
+            'create': True,
+            'collections': [(collection, mongoIndex)]}
+        mongoDB = MongoDB(**msOutDBConfig)
+        self.msDB = getattr(mongoDB, self.msConfig['mongoDB'])
+        self.dbColl = self.msDB[collection]
+
+    def create(self, pdict):
+        """
+        Create and return pileup data from campaigns dictionary
+
+        :param pdict: a dictionary representing MSPileup data
+        :return: list of MSPileupError or empty list
+        """
+        # first, create MSPileupObj which will be validated against its schema
+        try:
+            obj = MSPileupObj(pdict)
+            doc = obj.data
+        except Exception as exp:  # pylint: disable=broad-except
+            msg = f"Failed to create MSPileupObj, {exp}"
+            self.logger.exception(msg)
+            err = MSPileupError(pdict, msg)
+            return [err.DuplicateDocument()]
+
+        # insert document into underlying DB
+        try:
+            self.dbColl.insert_one(doc)
+        except errors.DuplicateKeyError:
+            msg = f"Failed to insert: {doc}, it already exist in DB"
+            self.logger.exception(msg)
+            err = MSPileupError(doc, msg)
+            return [err.DuplicateDocument()]
+        return []
+
+    def update(self, doc):
+        """
+        Update pileup data with provided input
+
+        :param doc: a dictionary of pieleup data to be updated
+        :return: list of MSPileupError or empty list
+        """
+        # check if given document contains pileup Name (unique key)
+        pname = doc.get('pileupName', '')
+        if not pname:
+            err = MSPileupError(doc, f'No document found for {pname}')
+            return [err.NoKeyFound()]
+
+        # look-up piileup document in underlying DB
+        spec = {'pileupName': pname}
+        results = self.get(spec)
+        if not results:
+            err = MSPileupError(spec, f'No document found for {spec} query')
+            return [err.NoDocumentFound()]
+
+        # at this point we should have single document corresponding to given pileup name
+        if len(results) != 1:
+            msg = f"Unique constrain violated for {pname}"
+            err = MSPileupError(spec, msg)
+            return [err.UniqueConstrainError()]
+
+        # we do not need to create MSPileupObj and validate it since our doc comes from DB directly
+        doc['lastUpdateTime'] = gmtime()
+        try:
+            self.dbColl.update_one(spec, {"$set": doc})
```

Comment:
How about adding a `self.logger.info("Pileup object %s successfully updated in the database", spec.get("pileupName"))`?

Reply:
done